dotnet/src/SemanticKernel.Core/Functions/KernelFunctionFactory.cs

```diff
@@ -25,15 +26,17 @@ public static class KernelFunctionFactory
     /// <param name="parameters">Optional parameter descriptions. If null, it will default to one derived from the method represented by <paramref name="method"/>.</param>
     /// <param name="returnParameter">Optional return parameter description. If null, it will default to one derived from the method represented by <paramref name="method"/>.</param>
     /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to use for logging. If null, no logging will be performed.</param>
+    /// <param name="metadataPropertyBag">Optional metadata in addition to the named values already provided in other arguments.</param>
     /// <returns>The created <see cref="KernelFunction"/> for invoking <paramref name="method"/>.</returns>
     public static KernelFunction CreateFromMethod(
         Delegate method,
         string? functionName = null,
         string? description = null,
         IEnumerable<KernelParameterMetadata>? parameters = null,
         KernelReturnParameterMetadata? returnParameter = null,
-        ILoggerFactory? loggerFactory = null) =>
-        CreateFromMethod(method.Method, method.Target, functionName, description, parameters, returnParameter, loggerFactory);
+        ILoggerFactory? loggerFactory = null,
+        ReadOnlyDictionary<string, object?>? metadataPropertyBag = null) =>
```

Comment:
Same comment as above - I would probably keep the naming simple and use just `metadata` or just `propertyBag`. 
Most probably it won't be possible to add method overload here, taking into account amount of optional parameters.

Reply:
Adopted `additionalMetadata` as suggested by @markwallace-microsoft